### PR TITLE
Kan 24 update the scan track split plugin to work with multiple calibrators

### PR DESCRIPTION
**Pull request by amadeuswi**
This PR is to push support for the UHF calibrator strategy to the main branch.

The differences wrt to the L-Band strategy are:
- Sometimes no calibrators are observed before or after a block. That happens when another block is observed directly before or after this one and the calibrators are shared.
- In addition to the single dish calibrators we now have several calibrators for interferometry in there too. These are weeded out in MuSEEK because they are not used.

The algorithm looks for a gap in the tracking data that is big enough to accommodate the scanning observation. This is used to split the tracking data into a part before and a part after the scanning. Then the iterator looks for 7 consecutive observations of the same calibrator with small shifts, i.e. the shifts up, left, down, right and to the center. This is identified as the single dish calibrator and the other calibrators are discarded.

play_after_scan_m000v

track_pointing_iterator_single_dish_calibrators_after_scan

### CODE REVIEW

**Comment by amadeuswi** [comment]
@spinemart suggested to include a skeleton calibration plugin to demonstrate usage of this functionality. Working on it...

**Review by spinemart — APPROVED** [review]

File: museek/util/track_pointing_iterator.py [hunk 1]
```diff
@@ -56,16 +70,81 @@ def iterate(self) -> Iterator[tuple[str, list[np.ndarray], np.ndarray]]:
...
+                start_end = (0, min(target_change_indices))
+            elif all(start_end[1] > target_change_indices):
+                start_end = (max(target_change_indices)+1, len(features)-1)
+
+            valid_index_range = range(start_end[0], start_end[1])
+            if max(features_diff[valid_index_range]) > upper:
+                print('WARNING: something went wrong')
+            result.append(valid_index_range)
+
+            plt.scatter(features[:-1], features_diff)
```
[the remark this answers is not in the document]
Comment (amadeuswi) [reply museek/util/track_pointing_iterator.py]: done

File: museek/util/track_pointing_iterator.py [hunk 2]
```diff
@@ -13,30 +15,37 @@ class TrackPointingIterator:
...
+                 plot_dir: str,
                  n_calibrator_observations: int = 2,
                  calibrator_observation_labels: list[str] = ('before_scan', 'after_scan'),
                  n_pointings: int = 5,
-                 n_centre_observations: int = 5,
-                 distance_threshold: float = 5.):
+                 n_centre_observations: int = 3,
+                 distance_threshold: float = 5.,
+                 pointing_slewing_thresholds: tuple[float, float] = (8., 15.),
+                 min_max_pointing_time: tuple[float, float] = (55., 80.)):
```
[the remark this answers is not in the document]
Comment (amadeuswi) [reply museek/util/track_pointing_iterator.py:27]: see docstring :)

[the remark this answers is not in the document]
Comment (amadeuswi) [reply museek/util/track_pointing_iterator.py:27]: the information there is limited, but it's hard to keep information like this up-to-date (very soon somebody may change the default values but nobody will be there to update the docstring... so I kept it relatively general)

**Pull request merged**